server/src/main/java/pl/masi/controllers/TestController.java

```diff
@@ -0,0 +1,48 @@
+package pl.masi.controllers;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import pl.masi.exceptions.AppException;
+import pl.masi.services.interfaces.ITestService;
+import pl.masi.entities.Test;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.List;
+
+@RestController(value = "/tests")
+public class TestController {
+
+    @Autowired
+    private ITestService testService;
+
```

Comment:
There is no api dedicated to create a new test. Is it deliberate?

Reply:
In this spring, we have to realize only 1-3 user stories. The functionality to add tests is not required there. This functionality may by quite demanding from UI point of view. Then I decided to implement api for adding test when it will be needed.